library/src/main/java/org/apache/apex/malhar/lib/wal/FSWal.java

```diff
@@ -0,0 +1,178 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.apex.malhar.lib.wal;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.EOFException;
+import java.io.IOException;
+
+import org.apache.hadoop.fs.FSDataOutputStream;
+
+import com.google.common.base.Preconditions;
+
+import com.datatorrent.lib.fileaccess.FileAccess;
+
+/**
+ * WAL implementation which allows writing entries to single file.
+ * the pointer type is the offset in the file. The entry is serialized into byte array and
+ * first length of the entry is written followed by the serialized bytes.
+ * @param <T>
+ */
+public class FSWal<T> implements WAL<T, Long>
+{
+  transient long bucketKey;
+  private transient String name;
+  private transient FileAccess fa;
+  private transient WAL.Serde<T> serde;
+
+  public FSWal(FileAccess fa, WAL.Serde<T> serde, long bucketKey, String name)
```

Comment:
My earlier comment just pertains to FSWal. I think bucket key is leftover from HDHT implementation of File System Wal.